tests/build/build.go

```diff
@@ -123,6 +128,241 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+				GinkgoWriter.Printf("component name: %s'\n", component.Name)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			AfterAll(func() {
+				err = f.AsKubeAdmin.CommonController.Github.DeleteRef(helloWorldComponentGitSourceRepoName, annotationsPurgeBranchName)
+				if err != nil {
+					Expect(err.Error()).To(ContainSubstring("Reference does not exist"))
+				}
+			})
+
+			It("initial build triggers a pipelinerun", func() {
+				timeout = time.Minute * 1
+				interval = time.Second * 1
+
+				Eventually(func() bool {
+					pipelineRun, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(annotationsComponentName, applicationName, testNamespace, "")
+					if err != nil {
+						GinkgoWriter.Println("PipelineRun has not been created yet")
+						return false
+					}
+					return pipelineRun.HasStarted()
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRun to start")
+
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "", 2)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			})
+
+			It("simple build can be triggered manually", func() {
+				component, err = f.AsKubeAdmin.HasController.GetComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				component.Annotations[controllers.BuildRequestAnnotationName] = controllers.BuildRequestTriggerSimpleBuildAnnotationValue
+				Expect(f.AsKubeAdmin.CommonController.KubeRest().Update(context.TODO(), component)).To(Succeed())
+
+				Eventually(func() bool {
+					pipelineRun, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(annotationsComponentName, applicationName, testNamespace, "")
+					if err != nil {
+						GinkgoWriter.Println("PipelineRun has not been created yet")
+						return false
+					}
+					return pipelineRun.HasStarted()
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRun to start")
+
+				component, err = f.AsKubeAdmin.HasController.GetComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(component.Annotations[controllers.BuildRequestAnnotationName]).To(BeEmpty(), "request annotation was not deleted")
+
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "", 2)).To(Succeed())
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			})
+
+			It("PaC provision is finished succesfully", func() {
+				component, err = f.AsKubeAdmin.HasController.GetComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				component.Annotations[controllers.BuildRequestAnnotationName] = controllers.BuildRequestConfigurePaCAnnotationValue
+				Expect(f.AsKubeAdmin.CommonController.KubeRest().Update(context.TODO(), component)).To(Succeed())
```

Comment:
observing this update component code pattern multiple times with this change.. will it be worth adding a new function named `UpdateComponent` inside `HasController`